src/common/lockdep.cc

```diff
@@ -290,17 +291,19 @@ int lockdep_will_lock(const char *name, int id, bool force_backtrace)
        p != m.end();
        ++p) {
     if (p->first == id) {
-      lockdep_dout(0) << "\n";
-      *_dout << "recursive lock of " << name << " (" << id << ")\n";
-      BackTrace *bt = new BackTrace(BACKTRACE_SKIP);
-      bt->print(*_dout);
-      if (p->second) {
-	*_dout << "\npreviously locked at\n";
-	p->second->print(*_dout);
+      if (!recursive) {
```

Comment:
Sorry I didn’t ask before merge, but is this conditional backwards? I was looking at this last week and it confused me.

Reply:
@liewegas ^

Reply:
bool recursive means it's a recursive mutex, which means that recursive locks are okay. so we only want to crash out if !recursive.